src/main/java/org/xbill/DNS/ResolverConfig.java

```diff
@@ -52,21 +52,25 @@
 ResolverConfig() {
 	if (findProperty())
 		return;
-	if (findSunJVM())
-		return;
+	boolean attemptedSun = false;
 	if (servers == null || searchlist == null) {
 		String OS = System.getProperty("os.name");
 		String vendor = System.getProperty("java.vendor");
 		if (OS.contains("Windows")) {
-			findWin();
-		} else if (OS.contains("NetWare")) {
-			findNetware();
-		} else if (vendor.contains("Android")) {
-			findAndroid();
-		} else {
-			findUnix();
-		}
+			if (findSunJVM() || findWin())
+				return;
+			else
+				attemptedSun = true;
```

Comment:
I don't think `attemptedSun` makes sense like this. It's only used once, so the code would be simpler to read if you just add

    else
        findSunJVM();

as the general fallback if none of the OS checks match.